universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -64,7 +64,17 @@ public boolean addTranslation(
     if (!errors.isEmpty()) {
       return ErrorAnd.error(errors);
     }
-    question = questionRepository.updateQuestionSync(new Question(definition));
+    switch (question.getLifecycleStage()) {
+      case DRAFT:
+        question = questionRepository.updateQuestionSync(new Question(definition));
+        break;
+      case DELETED:
+        return ErrorAnd.error(ImmutableSet.of(CiviFormError.of("Question %d was DELETED.")));
+      case ACTIVE:
+        // fallthrough
+      case OBSOLETE:
+        question = questionRepository.getOrCreateDraft(definition);
```

Comment:
Question (blocking): it seems odd that `update` is calling into a method called `getOrCreateDraft`. If it's `get`, does it update the existing one and then get it?